pom.xml

```diff
@@ -94,43 +76,13 @@
       <plugin>
         <groupId>org.jenkins-ci.tools</groupId>
         <artifactId>maven-hpi-plugin</artifactId>
-        <version>1.106</version>
         <configuration>
           <systemProperties>
             <hudson.bundled.plugins>${basedir}/work/plugins/ssh-slaves.hpl</hudson.bundled.plugins>
```

Comment:
[This hack](https://github.com/jenkinsci/jenkins/blob/e2dccc35572b64f5e602a6b1b3efb517c578872c/core/src/main/java/hudson/LocalPluginManager.java#L63) is probably obsolete. Try deleting the `maven-hpi-plugin` configuration, then doing a

    git clean -fdx && mvn hpi:run

and verify that http://localhost:8080/jenkins/pluginManaged/installed shows the this plugin as `1.11-SNAPSHOT` as expected.